core/mobile/src/backend/MobileAuthorizationBackend.ts

```diff
@@ -6,159 +6,30 @@
-  public issuerUrl?: string;
-  public static defaultRedirectUri = "imodeljs://app/signin-callback";
-  public get redirectUri() { return this.config?.redirectUri ?? MobileAuthorizationBackend.defaultRedirectUri; }
-  protected _baseUrl = "https://ims.bentley.com";
-  protected _url?: string;
-
-  public constructor(config?: MobileAppAuthorizationConfiguration) {
-    this.config = config;
-  }
-
-  /** Used to initialize the client - must be awaited before any other methods are called */
-  public async initialize(config?: MobileAppAuthorizationConfiguration): Promise<void> {
-    this.config = config ?? this.config;
-    if (!this.config)
-      throw new IModelError(AuthStatus.Error, "Must specify a valid configuration when initializing authorization");
-    if (this.config.expiryBuffer)
-      this.expireSafety = this.config.expiryBuffer;
-    this.issuerUrl = this.config.issuerUrl ?? this.getUrl();
-    if (!this.issuerUrl)
-      throw new IModelError(AuthStatus.Error, "The URL of the authorization provider was not initialized");
-
-    MobileHost.device.authStateChanged = (tokenString?: AccessToken) => {
-      this.setAccessToken(tokenString);
-    };
-
-    return new Promise<void>((resolve, reject) => {
-      assert(this.config !== undefined);
-      MobileHost.device.authInit({ ...this.config, issuerUrl: this.issuerUrl }, (err?: string) => {
-        if (!err) {
-          resolve();
-        } else {
-          reject(new Error(err));
-        }
-      });
-    });
-  }
-
-  /** Start the sign-in process */
-  public async signIn(): Promise<void> {
-    return new Promise<void>((resolve, reject) => {
-      MobileHost.device.authSignIn((err?: string) => {
-        if (!err) {
-          resolve();
-        } else {
-          reject(new Error(err));
-        }
-      });
-    });
-  }
-
-  /** Start the sign-out process */
-  public async signOut(): Promise<void> {
-    return new Promise<void>((resolve, reject) => {
-      MobileHost.device.authSignOut((err?: string) => {
-        if (!err) {
-          resolve();
-        } else {
-          reject(new Error(err));
-        }
-      });
-    });
-  }
-
-  public setAccessToken(token?: AccessToken) {
-    if (token === this._accessToken)
-      return;
-    this._accessToken = token;
-  }
+  protected _accessToken: AccessToken = "";
 
   public async getAccessToken(): Promise<AccessToken> {
-    if (!this._accessToken)
-      this.setAccessToken(await this.refreshToken());
-    return this._accessToken ?? "";
-  }
-
-  /** return accessToken */
-  public async refreshToken(): Promise<AccessToken> {
     return new Promise<AccessToken>((resolve, reject) => {
-      MobileHost.device.authGetAccessToken((tokenStringJson?: AccessToken, err?: string) => {
-        if (!err && tokenStringJson) {
-          resolve(tokenStringJson);
-        } else {
-          reject(new Error(err));
-        }
-      });
+      if (this._accessToken) {
+        resolve(this._accessToken);
```

Comment:
This will never fetch a new token once the old one has expired.

My advice in or call from weeks ago was to update MobileHost.device.authGetAccessToken (called below) to include the expiration date as a new parameter in the result. Technically, the code can parse the token to extract the expiration date, but I feel that that is a bad idea.

Once you know the expiration date, always fetch a new token if the current time is within 1 minute of the expiration date.

Reply:
Fixed, though I used ten minutes as the expiration buffer. This matches the electron auth behavior.